### ISSUE TITLE
Conversion to uff fails with dropout layers

### ISSUE THREAD

**Issue by ngeorgis** [ngeorgis/ca-fire-detector#1]
if you add a Dropout layer fails to create uff

```
x = base_model.output
x = GlobalAveragePooling2D()(x)
x = Dropout(0.45)(x)
predictions = Dense(num_classes, activation='softmax')(x)
```

**Comment by ngeorgis** [comment]
Seems that _optimize_for_inference.py_ to will not remove dropout layers. 

You have to edit your graph manually and delete all dropout layers (not an easy task) or else TensorRT frozen pb -> uff gets confused as in most cases dropout is just for training.

https://github.com/tensorflow/tensorflow/issues/5867

**Comment by shrutimittal90** [comment]
I am also facing issues due to dropout layer in converting frozen fine-tuned inception model to uff format required by Jetson.

**Comment by shrutimittal90** [comment]
Run training script -> export_inference_graph -> freeze_graph . It gets rid of all the extra nodes added during training

**Comment by ngeorgis** [comment]
Thank you @shrutim90 - will try this.

However, I cannot understand how this would be different from cell 32 in https://github.com/ngeorgis/ca-fire-detector/blob/master/fire-detection-jetson-save-ca.ipynb

where I run

`freeze_graph.freeze_graph('./fire_detector_tmp.pb', '', False, checkpoint_path, 'dense_1/Softmax',
                          "save/restore_all", "the_labels", 'frozen_fire_detector.pb', False,"")
`

**Comment by shrutimittal90** [comment]
@ngeorgis Freeze graphs just converts variable ops to constants, however export inference graph removes those additional nodes which create problem for uff conversion.

For me, uff conversion was working for inception graph released but when i retrained the same graph and freeze it as per this link: https://github.com/tensorflow/models/tree/master/research/slim#Pretrained , additional ops like Floor, RandomUniform, etc were present in the dropout layers which created problem for conversion.  The only additional step for released graph was export_inference_graph, and when I tried that for fine-tuned model, I could get rid of additional nodes and convert it to uff.

**Comment by ngeorgis** [comment]
No issues in the latest frameworks and SDKs.

**Comment by ngeorgis** [comment]
Fixed in https://github.com/ngeorgis/ca-fire-detector/commit/edd6e3cb1e3014234854034b7f869e39bfad2e2b
